File: src/rubika.js
const axios = require("axios");

const TOKEN = "CEDBIJ0IODNFFVMQLXUCPGSIJNSNCMAPTDHFIRRJZBXRIOZUYMAIDHURUAHZQNMN";

const BASE_URL = `https://messengerg2b1.iranlms.ir/v3/${TOKEN}`;

async function callMethod(method, params = {}) {
  try {
    const { data } = await axios.post(`${BASE_URL}/${method}`, params, {
      timeout: 15000,
    });
    return data;
  } catch (err) {
    const detail = err.response ? JSON.stringify(err.response.data) : err.message;
    console.error(`[rubika] خطا در متد ${method}:`, detail);
    throw err;
  }
}

function sendMessage(chatId, text, inlineKeypad = null) {
  const params = { chat_id: chatId, text };
  if (inlineKeypad) params.inline_keypad = inlineKeypad;
  return callMethod("sendMessage", params);
}

function sendFile(chatId, fileId, caption = "") {
  return callMethod("sendFile", {
    chat_id: chatId,
    file_id: fileId,
    text: caption,
  });
}

function updateBotEndpoint(url, type = "ReceiveUpdate") {
  return callMethod("updateBotEndpoint", { url, type });
}

function getMe() {
  return callMethod("getMe", {});
}

module.exports = {
  callMethod,
  sendMessage,
  sendFile,
  updateBotEndpoint,
  getMe,
};

File: rubika.py
import requests

TOKEN = "CEDBIJ0IODNFFVMQLXUCPGSIJNSNCMAPTDHFIRRJZBXRIOZUYMAIDHURUAHZQNMN"

BASE_URL = "https://botapi.rubika.ir/v3/" + TOKEN


def call_method(method, params=None):
    if params is None:
        params = {}
    url = BASE_URL + "/" + method
    response = requests.post(url, json=params, timeout=15)
    print("درخواست به:", url)
    print("وضعیت پاسخ:", response.status_code)
    print("متن خام پاسخ:", response.text)
    return response.json()


def send_message(chat_id, text, inline_keypad=None):
    params = {"chat_id": chat_id, "text": text}
    if inline_keypad is not None:
        params["inline_keypad"] = inline_keypad
    return call_method("sendMessage", params)


def send_file(chat_id, file_id, caption=""):
    params = {"chat_id": chat_id, "file_id": file_id, "text": caption}
    return call_method("sendFile", params)


def get_updates(limit=10, offset_id=None):
    params = {"limit": limit}
    if offset_id is not None:
        params["offset_id"] = offset_id
    return call_method("getUpdates", params)


def forward_message(from_chat_id, message_id, to_chat_id):
    params = {
        "from_chat_id": from_chat_id,
        "message_id": message_id,
        "to_chat_id": to_chat_id,
    }
    return call_method("forwardMessage", params)
